src/overlays/actors/ovl_Arms_Hook/z_arms_hook.c

```diff
@@ -245,7 +243,7 @@ void ArmsHook_Shoot(ArmsHook* this, PlayState* play) {
         sp60.z = this->unk1EC.z - (this->unk1E0.z - this->unk1EC.z);
         if (BgCheck_EntityLineTest1(&play->colCtx, &sp60, &this->unk1E0, &posResult, &poly, true, true, true, true,
                                     &bgId) &&
-            (func_800B90AC(play, &this->actor, poly, bgId, &posResult) == 0 ||
+            ((func_800B90AC(play, &this->actor, poly, bgId, &posResult) == 0) ||
```

Comment:
```suggestion
            (func_800B90AC(play, &this->actor, poly, bgId, &posResult) ||
```
Looks like it returns a boolean.

Reply:
Interestingly, all this function does is check if the floortype is of type 8 and nothing else. So in z_en_arrow, this function is called outside an if statement, meaning it does nothing. I went and added a comment to that use-case, let me know what you think:
`// `func_800B90AC` only returns a boolean, and does not process any code`